From 5f84b445686e8bcb5d1611da064184ea31b4d723 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 28 Apr 2020 14:42:14 +0300
Subject: [PATCH] [SW-12004] boot_desc: use img size from descriptor

Now that the descriptor has its image size inside, we can start using
it.
Replace usage of the hard coded size and use the one provided by the
image descriptor.

Change-Id: Icfcc6764f9b8bf3ff46f47480aa07ede397b24b5
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ppboot.c b/app/ppboot.c
index 8493ea5..67062e6 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -43,6 +43,7 @@
 
 	desc->spi_offset = next_image_base_offset(next_img_off, name) +
 						HL_SPI_LEGACY_PREBOOT_OFF;
+	desc->img_size = PREBOOT_SIZE;
 	return image_pending;
 }
 
@@ -76,10 +77,11 @@
 		preboot_puts_raw("candidate");
 
 	preboot_puts_hex_raw(" (offset: ", desc.spi_offset, false);
+	preboot_puts_hex_raw(", size: ", desc.img_size, false);
 	preboot_puts(")\n");
 
 	spi_copy((void *)PREBOOT_TEXT_BASE,
-		 desc.spi_offset, PREBOOT_SIZE, name);
+		 desc.spi_offset, desc.img_size, name);
 }
 
 static void rst_src_dump(void)
diff --git a/app/preboot.c b/app/preboot.c
index 5d3d99e..1192e68 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -85,6 +85,7 @@
 		base_off = modifiable_image_base_offset(&pending);
 
 	desc->spi_offset = base_off + HL_SPI_LEGACY_UBOOT_OFF;
+	desc->img_size = UBOOT_SIZE;
 	return pending;
 }
 
@@ -121,9 +122,10 @@
 	preboot_puts_raw("Preparing to run U-Boot ");
 	preboot_puts_raw(uboot_image);
 	preboot_puts_hex_raw(" (offset: ", desc.spi_offset, false);
+	preboot_puts_hex_raw(", size: ", desc.img_size, false);
 	preboot_puts(")\n");
 
-	spi_copy((void *)UBOOT_TEXT_BASE, desc.spi_offset, UBOOT_SIZE, name);
+	spi_copy((void *)UBOOT_TEXT_BASE, desc.spi_offset, desc.img_size, name);
 }
 
 #ifndef TARGET_IS_TESTER
diff --git a/common/boot_desc.c b/common/boot_desc.c
index 96e5745..033b588 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -101,6 +101,7 @@
 
 		preboot_puts_hex(" - found! using index: ", idx, false);
 		desc->spi_offset = img_desc[i].offset[idx];
+		desc->img_size = img_desc[i].size[idx];
 		return idx;
 	}
 
diff --git a/common/boot_desc.h b/common/boot_desc.h
index c6485a6..3c8c19d 100644
--- a/common/boot_desc.h
+++ b/common/boot_desc.h
@@ -15,6 +15,7 @@
 
 struct boot_desc {
 	u32 spi_offset;
+	u32 img_size;
 };
 
 /*
